src/devices/MatrixKeyboard/MatrixKeyboard.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Timers;
+
+namespace Iot.Device.MatrixKeyboard
+{
+    /// <summary>
+    /// GPIO Matrix Keyboard Driver
+    /// </summary>
+    public class MatrixKeyboard : IDisposable
+    {
+        /// <summary>
+        /// Get row pins
+        /// </summary>
+        public IEnumerable<int> RowPins => rowPins;
+
+        /// <summary>
+        /// Get column pins
+        /// </summary>
+        public IEnumerable<int> ColPins => colPins;
+
+        /// <summary>
+        /// Get or set scanning frequency in Hertz
+        /// </summary>
+        public double ScanFreq
+        {
+            get => 1000 / (scanTimer.Interval / colPins.Length);
+            set => scanTimer.Interval = 1000 / (value * rowPins.Length);
+        }
+
+        /// <summary>
+        /// Initialize keyboard
+        /// </summary>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="rowPins">Row pins</param>
+        /// <param name="colPins">Column pins</param>
+        /// <param name="scanFreq">Scanning frequency</param>
+        public MatrixKeyboard(IGpioController gpioController, IEnumerable<int> rowPins, IEnumerable<int> colPins, double scanFreq = 50)
+        {
+            gpio = gpioController;
+
+            this.rowPins = rowPins.ToArray();
+            this.colPins = colPins.ToArray();
+            buttonValues = new PinValue[this.rowPins.Length * this.colPins.Length];
+
+            scanTimer = new Timer(1000 / (scanFreq * this.rowPins.Length));
+            scanTimer.Elapsed += ScanTimerElapsed;
+        }
+
+        /// <summary>
+        /// Open GPIO pins then start keyboard scanning
+        /// </summary>
+        public void StartScan()
```

Comment:
You should add checks here, like for example, what happens if we already called StartScan? What happens if rowPins is empty, what happens if colpins is empty?